docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,128 @@
+"apply" is defined later), and more than one file is defining the same value,
+.editorconfig file listed first in the command arguments takes precedence over
+later files.
+
+# Discovery of .editorconfig files in MSBuild
+
+Discovery of .editorconfig files is not done by csc.exe or vbc.exe, but rather
+the host.  For managed code projects, a new build task,
+DiscoverEditorConfigFiles, is introduced which runs immediately before the
+CoreCompile target so all inputs are found.  The inputs to the task are:
+
+- InputFiles: a list of ITaskItems which are the input files from which we must
+  locate .editorconfig files in parent directories. This *MUST* include Compile
+  and AdditionalFile items, and potentially others.
+- RootDirectories: an optional list of root directories which the build task
+  *MUST NOT* ascend higher during any walking. This is a list because a build
+  environment that segregates intermediate directories or build directories
+  outside the root of the source tree might wish to ensure that no walking is
+  ever done outside the source path, or intermediate/output directories, none
+  of which may be a parent of any other root (or indeed on Windows might not
+  even live on the same drive!)
+
+The outputs of this task are:
+
+- EditorConfigFiles: a list of ITaskItems which are paths to discovered
+  .editorconfig files. These *MAY* be relative paths to the current working
+  directory, in the same way that input files *MAY* be relative paths.
+
+The default target maps these items to an EditorConfig item group, that is
+consumed by the CoreCompile target.
+
+A boolean property DiscoverEditorConfigFiles *MUST* be provided which
+conditions the running of the DiscoverEditorConfigFiles target in the first
+place. If unset, the target behaves as if this is set to true.
+
+The output of this task *MUST* be deterministic relative to the inputs.
+Ordering changes (even if there is no semantic difference) is explicitly
+disallowed.
+
+# Interpretation of the contents of an .editorconfig file
+
+.editorconfig files are parsed per the [.editorconfig spec](http://editorconfig.org/#file-format-details).
+
+The compilers as of this writing *MUST NOT* assign any meaning to any
+.editorconfig key or value and apply it to core compiler behavior, to mitigate
+backwards compatibility concerns about existing .editorconfig files. The
+previous statement is scoped to allow for the possibility that in future
+versions the compiler *MAY* consume .editorconfig options to control other
+settings (such as defaults for warnings/errors) but the spec will be clarified
+at such a time. The expectation is such a change happens either at a major
+version where breaking changes are acceptable, or controlled under a /langver
+or equivalent switch.
+
+# PathMap
+
+*This section is informational only and does not propose any new behavior.*
+
+As a reminder, the `/pathmap` flag to the compiler allows the compiler to "rewrite" paths prior to them being
```

Comment:
how will this ordering and pathmap thing will work in IDE? will .editorconfig part of solution as some kind of text document? will Workspace provide an API that hide all these from IDE consumer? if .editorconfig is part of solution and has pathmap translated path already applied, is there a way to get actual file path from .editorconfig file?